src/electron/main.tsx

```diff
@@ -135,7 +132,8 @@ if (!handleSquirrelEvent()) {
         webPreferences: {
           preload: `${__dirname}/preload.js`,
           nodeIntegration: true,
-          enableRemoteModule: true,
+          contextIsolation: false,
```

Comment:
can you enable contextIsolation?

Reply:
If context isolation is enabled then node intergation is disabled

Reply:
Which is definitely needed

Reply:
are we able to do that in a different way without disabling contextIsolation?

Reply:
I believe so but I need to look into it

Reply:
> are we able to do that in a different way without disabling contextIsolation?

it will take me a bit

Reply:
> are we able to do that in a different way without disabling contextIsolation?

we are not able to as enabling context isolation disables node integration (import) for the main renderer.
It is still more secure then before as remote is restricted

Reply:
@seeden let me know if that answers your questions

Reply:
we can use contextBridge https://www.electronjs.org/docs/tutorial/context-isolation#migration

`
const { contextBridge } = require('electron')

contextBridge.exposeInMainWorld('myAPI', {
  doAThing: () => {}
})
`

Reply:
Yep  i have already tried that,

Reply:
> Yep i have already tried that,

i have it stashed but every file in the whole project will need to get its imports removed.
Its simply not feasible

Reply:
Let me make a new branch with it

Reply:
ok

Reply:
@seeden https://github.com/jack60612/chia-blockchain-gui/tree/testcontext